### PR TITLE
chore: move turborepo tests to primary workflow

### PR DESCRIPTION
**Pull request by chris-olszewski**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -91,8 +91,16 @@ jobs:
             cli/**
             packages/**
             shim/**
+            Cargo.*
+            .cargo/**
             !**.md
             !**.mdx
+      - name: Test changes
+        id: tests
```
Comment (sokra) [inline .github/workflows/test.yml:99]: Maybe make it a little bit more specific, like `cli_tests` or `go_tests` or `repo_tests`

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -91,8 +91,16 @@ jobs:
...
+        uses: technote-space/get-diff-action@v6
+        with:
+          PATTERNS: |
+            cli/integration_tests/**/*.t
```
Comment (sokra) [inline .github/workflows/test.yml:103]:
```suggestion
            cli/integration_tests/**
```

There are some `*.sh` files or `turbo.json`s in these tests that should probably also trigger a test run

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -460,6 +510,7 @@ jobs:
     if: needs.determine_jobs.outputs.rust == 'true'
     strategy:
       fail-fast: false
+      runs-on: ${{ matrix.os }}
```
Comment (sokra) [inline .github/workflows/test.yml:513]: There is a `runs-on: ${{ matrix.os.runner }}` a few lines below.

Also there is no `runs-on` in `strategy`

File: .github/actions/build-debug-turborepo/action.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  target:
+    description: "Compilation target"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3
+    - uses: ./.github/actions/setup-node
+    - uses: ./.github/actions/setup-go
+    - uses: actions-rs/toolchain@v1
```
Comment (sokra) [inline .github/actions/build-debug-turborepo/action.yml:13]: Give these steps names

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -271,6 +280,47 @@ jobs:
...
+          path: |
+            target/debug/turbo
+            target/debug/turbo.exe
+
+  turborepo_integration_tests:
+    name: Cram integration tests
+    if: needs.determine_jobs.outputs.go == 'true' || needs.determine_jobs.outputs.tests == 'true'
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```
Comment (sokra) [inline .github/workflows/test.yml:308]:
```suggestion
    if: needs.determine_jobs.outputs.go == 'true' || needs.determine_jobs.outputs.tests == 'true'
    strategy:
      matrix:
        os:
          - name: ubuntu
            runner: ubuntu-latest
          - name: macos
            runner: macos-latest
    runs-on: ${{ matrix.os.runner }}
    name: Cram integration tests on ${{ matrix.os.name }}
```

You could switch to `runner: ubuntu-latest-16-core-oss` if you want this to be faster, but it costs ~$3/h

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -271,6 +280,47 @@ jobs:
           command: check
           args: --workspace --all-targets
 
+  build_turborepo:
+    name: Build Turborepo
+    needs: [determine_jobs, rust_prepare]
+    if: needs.determine_jobs.outputs.go == 'true' || needs.determine_jobs.outputs.tests == 'true' || needs.determine_jobs.outputs.node == 'true'
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```
Comment (sokra) [inline .github/workflows/test.yml:289]:
```suggestion
    needs: [determine_jobs, rust_prepare]
    if: needs.determine_jobs.outputs.go == 'true' || needs.determine_jobs.outputs.tests == 'true'
    strategy:
      matrix:
        os:
          - name: ubuntu
            runner: ubuntu-latest
          - name: macos
            runner: macos-latest
          - name: windows
            runner: windows-latest
    runs-on: ${{ matrix.os.runner }}
    name: Build Turborepo for ${{ matrix.os.name }}
```

Seems like `runs-on` is missing here?

You could switch to `runner: ubuntu-latest-16-core-oss` if you want this to be faster, but it costs ~$3/h

File: .github/workflows/test.yml (hunk 3 above)
Comment (sokra) [inline .github/workflows/test.yml:513]:
```suggestion
```

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -271,6 +280,47 @@ jobs:
           command: check
           args: --workspace --all-targets
 
+  build_turborepo:
+    name: Build Turborepo
+    needs: [determine_jobs, rust_prepare]
+    if: needs.determine_jobs.outputs.go == 'true' || needs.determine_jobs.outputs.tests == 'true' || needs.determine_jobs.outputs.node == 'true'
```
Comment (sokra) [inline .github/workflows/test.yml:286]: Why does this need to run when `node` has changed? If this depends on some JS files you can add these files to the `go` diff and remove that

**Review by sokra — CHANGES_REQUESTED** [review]

**Comment by sokra** [comment]
Also make sure to remove the old workflow files in this PR too

**Comment by chris-olszewski** [comment]
Closing in favor of #2854

**Pull request closed without merging**